orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/GermlineDeletionUtil.java

```diff
@@ -0,0 +1,193 @@
+    {
+        for(ExonData exon : transcript.exons())
+        {
+            if(!deletionsCoverExon(deletionsForGene, exon))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @NotNull
+    public static TranscriptData findCanonicalTranscript(@NotNull String geneNameToFind, @NotNull EnsemblDataCache ensemblDataCache)
+    {
+        GeneData gene = ensemblDataCache.getGeneDataByName(geneNameToFind);
+        if(gene == null)
+        {
+            throw new IllegalStateException("Could not find gene in ensembl data cache with name: " + geneNameToFind);
+        }
+
+        TranscriptData transcript = ensemblDataCache.getCanonicalTranscriptData(gene.GeneId);
+        if(transcript == null)
+        {
+            throw new IllegalStateException("Could not find canonical transcript in ensembl data cache for gene with id: " + gene.GeneId);
+        }
+
+        return transcript;
+    }
+
+    @NotNull
+    public static String getChromosome(@NotNull List<GermlineDeletion> deletions)
+    {
+        Set<String> chromosomes = deletions.stream().map(d -> d.Chromosome).collect(Collectors.toSet());
+        if(chromosomes.size() == 1)
+        {
+            return chromosomes.iterator().next();
+        }
+        else if(chromosomes.isEmpty())
+        {
+            throw new IllegalArgumentException("No germline deletions provided to get chromosome from.");
+        }
+        else
+        {
+            throw new IllegalStateException("Germline deletions disagree on chromosome.");
+        }
+    }
+
+    @NotNull
+    public static String getChromosomeBand(@NotNull List<GermlineDeletion> deletions)
+    {
+        Set<String> chromosomeBands = deletions.stream().map(d -> d.ChromosomeBand).collect(Collectors.toSet());
+        if(chromosomeBands.size() == 1)
+        {
+            return chromosomeBands.iterator().next();
+        }
+        else if(chromosomeBands.isEmpty())
+        {
+            throw new IllegalArgumentException("No germline deletions provided to get chromosome band from.");
+        }
+        else
+        {
+            throw new IllegalStateException("Germline deletions disagree on chromosome band.");
+        }
+    }
+
+    @NotNull
+    public static boolean getReportedStatus(@NotNull List<GermlineDeletion> deletions)
+    {
+        Set<Boolean> reportedStatuses = deletions.stream().map(d -> d.Reported).collect(Collectors.toSet());
+        if(reportedStatuses.size() == 1)
+        {
+            return reportedStatuses.iterator().next();
+        }
+        else if(reportedStatuses.isEmpty())
+        {
+            throw new IllegalArgumentException("No germline deletions provided to get reported statuses from.");
+        }
+        else
+        {
+            throw new IllegalStateException("Germline deletions disagree on reported statuses.");
+        }
```

Comment:
I'm pretty sure it shouldn't be able to, since I'm pretty sure all germline deletions cover at least part of an exon.

However, there is no reason to rely on this, or to assume that this won't be changed in Purple in the future. So, I have changed it such that reported is true when at least one of the germline deletions that was merged into the LOH or GainLoss object was reportable.